### PR TITLE
802 Wrap tag around when hits MPI max tag (attribute MPI_TAG_UB)

### LINKED ISSUE #802: Use MPI constant for max allowed tag
**Describe the bug**
Currently we don't use the proper MPI constant and tags monotonically increase.

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #802

### CODE REVIEW

**Comment by lifflander** [comment]
These tests will all fail until checkpoint changes get merged!

File: src/vt/messaging/active.cc [hunk 1]
```diff
@@ -322,7 +323,19 @@ ActiveMessenger::SendDataRetType ActiveMessenger::sendData(
 ) {
   auto const& data_ptr = std::get<0>(ptr);
   auto const& num_bytes = std::get<1>(ptr);
-  auto const send_tag = tag == no_tag ? cur_direct_buffer_tag_++ : tag;
+
+  int send_tag = 0;
+  if (tag == no_tag) {
```
Comment (PhilMiller) [inline src/vt/messaging/active.cc:328]: You reversed the sense of the comparison here - should be `tag != no_tag`

File: src/vt/utils/mpi_limits/mpi_max_tag.h [hunk 2]
```diff
@@ -53,30 +53,40 @@
...
 
 /**
- * \struct MPI_MaxTag
+ * \struct MPI_Attr
  *
- * \brief Get the MPI maximum value for a tag. This can vary across MPI
- * implementations, but the MPI standard dictates it be at least 32767.
+ * \brief Get MPI attributes, including the MPI maximum value for a tag. This
+ * can vary across MPI implementations, but the MPI standard dictates it be at
+ * least 32767.
```
Comment (PhilMiller) [inline src/vt/utils/mpi_limits/mpi_max_tag.h:60]: This comment is partially out of place now. The bits about the tag should be on `getMaxTag`

File: src/vt/messaging/active.cc [hunk 3]
```diff
@@ -322,7 +323,19 @@ ActiveMessenger::SendDataRetType ActiveMessenger::sendData(
...
+    send_tag = tag;
+  } else {
+    auto const max_tag = util::MPI_Attr::getMaxTag();
+    send_tag = cur_direct_buffer_tag_++;
+
+    // If max, wrap around back to the starting tag
+    if (cur_direct_buffer_tag_ >= max_tag) {
+      cur_direct_buffer_tag_ = starting_direct_buffer_tag;
+    }
+  }
```
Comment (PhilMiller) [inline src/vt/messaging/active.cc:338]: I think we may need to be a bit more cautious here about incrementing through `INT_MAX` if `MPI_TAG_UB` is equal to it.

I think what we want is rather

```
auto const max_tag = util::MPI_Attr::getMaxTag();
if (cur_direct_buffer_tag_ == max_tag) {
      cur_direct_buffer_tag_ = starting_direct_buffer_tag;
    }
send_tag = cur_direct_buffer_tag_++;
```

Comment (lifflander) [reply src/vt/messaging/active.cc:337]: Ok, fixed.

File: src/vt/utils/mpi_limits/mpi_max_tag.h (hunk 2 above)
Comment (lifflander) [reply src/vt/utils/mpi_limits/mpi_max_tag.h]: Fixed

File: src/vt/messaging/active.cc (hunk 1 above)
Comment (lifflander) [reply src/vt/messaging/active.cc]: Fixed

**Pull request merged**